tests/unit/test_lib_per_app_requires.py

```diff
@@ -127,6 +128,7 @@ def test_ipa_events(self):
         # WHEN an ingress relation is formed
         # THEN the ready event is emitted
         with capture(self.harness.charm, IngressPerAppReadyEvent):
+            self.harness.add_network("10.0.0.10")
```

Comment:
can we add a test where we call provide_ingress_requirements without an IP, and there is no network, and assert that there is an `ip=null` in the databag?

Reply:
Yes, I have added another test case to ensure the ip field in unit databag is `null`.